netlify/remix-compute
# feat: add @netlify/vite-plugin-react-router to support React Router 7

## Description

Remix has converged with React Router and the "next version" of Remix was released as React Router 7.

This PR adds a new package to support React Router 7: `@netlify/vite-plugin-react-router`. The naming convention is different as this is purely a Vite plugin (previous "adapters" also supported the Remix Classic Compiler), so we were able to follow the [documented naming convention](https://vite.dev/guide/api-plugin#conventions).

## Future work

- This does not yet add support for edge rendering.
- An upgrade/migration guide from Remix 2 is coming soon
- The plugin source is 95% duplicated from `@netlify/remix-adapter`. We should probably refactor to extract the shared logic.

## Related Tickets & Documents

FRB-1511

Supersedes https://github.com/netlify/remix-compute/pull/471

See also https://github.com/netlify/remix-template/issues/157

## QA Instructions, Screenshots, Recordings

This PR adds extensive e2e tests.

I also added a new React Router Netlify template here: https://github.com/netlify/react-router-template/pull/1/. It uses this new plugin, and you can see the successful deploy preview there.
